content/4_Gold/String_Hashing.mdx

```diff
@@ -167,6 +184,21 @@ public static long get_hash(int a, int b) { // Endpoints of the substring
 
 </LanguageSection>
 
+This implementation calculates
+$$
+\texttt{hsh}[i] = \left(\sum_{x = 0}^i P^{i - x} \cdot S[x]\right) \bmod M
+$$
+
+The hash of any particular substring $S[a : b]$ is then calculated as
+$$
+\left(\sum_{x = a}^b P^{b - x} \cdot S[x] \right) \bmod M = (\texttt{hsh}[b] - \texttt{hsh}[a] \cdot P^{b - a + 1}) \bmod M
+$$
+using prefix sums. This is nice because the highest power of $P$ in that polynomial
+will always be $P^{b - a}$.
+
+Since $10^9 + 9$ and $9973$ are both prime, the probability of collision when using this hash
+is at most $\frac{N}{10^9 + 9} < 10^{-4}$, by the Schwarz-Zippel lemma.
```

Comment:
Specify what the probability of collision means? If you select any two strings of length `<= N` and a random base modulo $10^9+9$ the probability that they hash to the same thing is at most $10^{-4}$.